core/src/main/scala/org/apache/spark/ui/SparkUI.scala

```diff
@@ -120,7 +120,7 @@ private[spark] class SparkUI private (
         attemptId = None,
         startTime = new Date(startTime),
         endTime = new Date(-1),
-        duration = 0,
+        duration = System.currentTimeMillis() - startTime,
```

Comment:
This should not be set unless endtime is.

Reply:
![1](https://user-images.githubusercontent.com/26266482/31758998-5fc3d946-b4e2-11e7-9839-44b1fceb63bf.png)

I would like to ask, running the spark application in master ui,  why real-time display how long?
The logic of these two places I think should be the same.

Reply:
I'm neutral to this change. Normally we don't update the duration until the attempt has been finished. But you can also argue that we do in other places.

Reply:
does this have an effect on the UI?

Reply:
No effect.
In master page, ‘duration ’is calculated using the duration attribute of the ApplicationInfo object.
![1](https://user-images.githubusercontent.com/26266482/32168880-2925daf4-bda9-11e7-8e89-925d5d29bf2c.png)

![2](https://user-images.githubusercontent.com/26266482/32168885-2e6c9aca-bda9-11e7-9f77-085c4721ead3.png)

And ApplicationAttemptInfo object does not matter.

Reply:
ok seems it's better to make them consistent, LGTM

Reply:
I'm on the fence on this change, I would tend to agree with @srowen but as @cloud-fan said, the consistency with the UI would make sense. If we did keep this change I thing removing the redundant code would also make sense.